Improves janicart and adds secway

paprka opened the pull request:
**Neither of these are on the map yet**
- Janicart can no longer deflect bullets
- Janicart cannot hold reagents
- Janicart is no longer sanic, about as fast as someone on green (roundstart) nutrition
- Janicart is no longer immune to lube
- Janicart can no longer move on tiles without gravity
- Janicart can be upgraded with a 'floor buffer' which makes it function like the janiborg does
- Added a 'secway' (thanks mrperson for the name) which is more or less a variant of the janicart with no cool functions other than you can be a fatass cop and not suffer slowdown on it (people on blue nutrition will still outrun secways)

Fixes https://github.com/tgstation/-tg-station/issues/7241

![cool1](https://cloud.githubusercontent.com/assets/7247232/5755250/1e716c48-9c57-11e4-93f2-9c55e5f07fc3.gif)

![cool2](https://cloud.githubusercontent.com/assets/7247232/5755252/20324d7c-9c57-11e4-972d-61245b009413.gif)

Sprite credit to the segway mostly to paradise (I'm pretty sure we mocked this up ages ago and they took them anyway?)

Note: I haven't tested this in zero-g environments or space, because I don't know how to disable the grav gen (bomb it?) so if someone could do that it would be swell.

Commit:
improves janicart and adds secway

paprka commented:
Pretty unsure of how to fix the attack animation breaking the pixel placement

Jalleo commented:
have the segway use the attack animation as well somehow?

Commit:
fixes secway sprite

TZK13 commented:
The janicart doesn't seem to be very useful for actually cleaning up without the floor buffer if you can't wet your mop in it, why not just make the floor buffer default?

Besides that I rather like this.

paprka commented:
it can hold your trashbag and you can use it with the backpack floor mister with no slowdown

come on get creative dummy

Commit:
another fix for the janicart and secway

pudl commented:
can i make a better segway sprite

Steelpoint commented:
Secway seems interesting, but more of a gimmick.

Love the Janicart though.

GunHog commented:
Love these both :+1: Btw, to disable the gravigen, just delete it (fastest), go to the chamber and turn it off, or turn the APC off. How do you acquire the floor buffer attachment? I do not see a way to get it in your code yet.

RemieRichards commented:
Should probably wait for https://github.com/tgstation/-tg-station/pull/7049

Fox-McCloud commented:
:-1: 

Even when you fix the the ability of these to be impacted by bullets, these still just become a powergamey/griefy tool; blocking areas, etc. Everyone and their brother who has a weapon wants one because it makes you incredibly difficult to both due to speed benefits it has(ie: no slowdown) and its impact on hitboxes.

It also sets the precedence for other jobs to want vehicles as well---cargo, medical, science, and engineering all come to mind.

Steelpoint commented:
To be honest I see the Security Segway as a perverted attempt to change Security's image to that of mall cops, nor do I see it as being anything more than a waste of space in the brig.

I dislike the Secway, although I still like the new Janicart.

Miauw62 commented:
>a device that makes you not suffer slowdown
>waste of space

ayy lmao

Steelpoint commented:
I would prefer if the sprite was something more fitting than a item used by fat mall cops.

Ultimantly I just dislike the visual image it presents for security.

GunHog commented:
@Steelpoint get that sprited, it sounds amazing!

dannno commented:
that sounds dumb lol
I do kind of agree though, secways seem a bit goofy for security.

Steelpoint commented:
>asking me to sprite

Here's a 5 minute mock up that represents the sprite of what I'm looking for.

![screenshot 2015-01-16 01 26 16](https://cloud.githubusercontent.com/assets/6595389/5762672/c47a46d8-9d1e-11e4-90cf-08df88913b7d.png)

Jalleo commented:
Why does that image make me feel you are then going to try and buff it later to become the complete madmans device to get around the station while low on health?

Miauw62 commented:
I DISAGREE WITH THIS REIMAGINING OF SECURITY AS A PARAMILITARY ORGANIZATION.

Miauw62 commented:
imo secways being a bit goofy makes them even better.

Steelpoint commented:
They are a private security force not some bumbling overweight mall cops Miauw. Look, its just a suggestion as I think the segway is a bit goofy for sec. 

Why not make it a public access segway in arrivals, or make it that only someone that is "fat" can get on and use it, rename it as the "Fatway".

Jalleo commented:
we dont have fatsprites at the moment sorry.

dannno commented:
you can still be fat tho

Lo6a4evskiy commented:
What's wrong with it holding reagents? How are you supposed to wet your trusty mop?

Otherwise merge yesterday.

Oh, add police siren to cop wagon.

paprka commented:
steelpoint can you not turn another feature into your fucking retarded security obsessed rant soapbox thanks

it isn't even on the map and probably never will be, it's only an admin spawnable thing just like the janicart for now

i mostly just don't want to deal with map merger stuff since this PR will probably be around a while

ike709 commented:
honestly I like them both ( :+1: ) except for the janicart not being able to hold reagents (atleast water pls). Having a floor buffer by default would be nice but not gamebreakingly necessary.

paprka commented:
janicart not holding reagents is so that the current janicart is not way outclassed by this new one, and you can choose to use one or the other. we have two janitors now. janitor number 1 would use janicart + backpack sprayer or floor buffer because he's hot shit, janitor 2 would roll around with the old fashioned one and mop like regular. again, you might think janitor 1 is objectively superior at cleaning compared to 2, but keep in mind that the amount of times the janitor has his cart spaced by the mime will be multiplied by a million if you roll around in the pussy wagon. it breeds jealousy.

GunHog commented:
Are you making a PR to add these to the map if this one is merged?

paprka commented:
probably just the janicart since it's balanced now, but the secway is up to the mappers

it's mostly meant to be an admin spawn for good goy- i mean officers

Commit:
improves the sprites and code a bit

peoplearestrange commented:
+1 to the janicart. Though I agree that its not a huge benefit without the scrubber. Maybe it could be faster without it? Or slower with it?

As for the sec thing... Really not sure on that, it just seems a little gimmicky and kinda makes sec seem more like a laughing stock than they already are.

paprka commented:
too lazy to wait for remie's thing

paprka commented:
nvm

Commit:
Merge branch 'master' of https://github.com/tgstation/-tg-station into readdjanikart

Conflicts:
	code/game/objects/structures/janicart.dm

paprka commented:
conflicts fixed

paprka commented:
Fixed https://github.com/tgstation/-tg-station/issues/7241 in this

RemieRichards commented on code/game/objects/structures/janicart.dm:
```diff
@@ -211,31 +229,53 @@
+		"<span class='notice'>[M] climbs onto the [callme]!</span>",\
+		"<span class='notice'>You climb onto the [callme]!</span>")
 
+	M.buckled = src
```
Nice "fix" by duplicating everything.

paprka replied:
You mean not removing everything like you did, which broke it..................?

RemieRichards replied:
You mean the shit that's already at the base buckling hence why I removed it, yes.

paprka replied:
we don't need the base buckling for this we need user_buckle_mob that you added

I tried it your way, you broke it, I reverted it, it works without bugs

Not my problem

RemieRichards replied:
you realise user_buckle_mob calls buckle_mob, you absolute tool.

paprka replied:
i really don't care dude, this works and if you want to improve it you can do it yourself and actually test this time

RemieRichards replied:
Duplicated code won't get in. and if you're not going to fix it then there's no point even working on fixing the janicart.

paprka replied:
>break something with a new system
>you have to fix it!

you're a fucking asshole you know that remie

RemieRichards replied:
you can't break something that was already broken, you're revamping the janicart, it's your issue.
>kludging some horrible code
it's a fix guys!

paprka replied:
IT WASN'T ALREADY BROKEN

IT WORKED PERFECTLY FINE

I IMPROVED THE BALANCING AND YOU TOLD ME TO WAIT FOR YOUR PR

I DID, IT COMPLETELY BROKE IT

HOW IS THIS MY RESPONSIBILITY FOR YOUR UNTESTED CODE?

RemieRichards replied:
I'm not testing an unused object. get over yourself.

paprka replied:
you mean an unused object you already knew i was readding

cool remie thanks for being a considerate coder and not someone who just casually fucks up things for other people

RemieRichards replied:
yeah because I totally shouldn't expect the person who's working on the janicart to actually be a good coder capable of fixing an issue without copy paste.

paprka replied:
I totally shouldn't expect the maintainer to know how to refactor without breaking things

RemieRichards replied:
didn't break shit, it was broken already. if I'd have known it was broken because of me I'd of fixed it, except the first I hear about it is some idiot copy pasting bad code and then getting pissy at me. you can see why I'd rather not work with someone like that.

paprka replied:
No, the janicart worked perfectly fine before your update, it was just unused. You broke it, and now you're being a moron about it. Real professional dickhead.

RemieRichards replied:
>being a moron about it

Where was I a moron? you're the one being a dick, for literally no reason.

RemieRichards commented:
Duplicated Code.

paprka commented:
Autistic shitcoder proves giacom right by breaking code and yelling at people: the PR

RemieRichards commented:
Calling me Autistic now? Note to self, Singulo was right about bloopers.

AlexanderUlanH commented:
Wow, something unintended happened as a result of a change to some of the code and now some code that was written before that change has to be written differently to use it correctly.  This is unprecedented.

paprka commented:
Are you fucking joking Alexander?

I never copypasted anything. I reverted the code that remie changed that broke janicarts. That's it. If remie broke them, it's on him, I'm not here to fix bugs with anything that other coders (a maintainer no less) creates with their bad refactoring.

RemieRichards commented:
>on him, 

Oh you've got to be fucking with me now, I'm female, get it right you shitbag.

and no I didn't copy paste it, I *removed* copy paste from all buckling objects by putting it in one place, that's called a refactor @paprka but you wouldn't know how to do one of those.

paprka commented:
Do you break things in refactors too and then yell at people who try to correct it instead of offering a solution, and tag their PRs as do not merge?

Maybe I'm just new at this xD

RemieRichards commented:
Except you didn't correct it, you made it worse, so I pointed that out, and then explained why you made it worse, you never once offered to actually fix it or cared to.

paprka commented:
GUYS I MADE THE BUG WORSE BY FIXING IT!!!

RemieRichards commented:
YOU DIDN'T FIX IT, YOU COPY PASTED OLD, OBSOLETE CODE, EFFECTIVELY SNOWFLAKING THIS SHIT UP.

paprka commented:
Uh you don't buckle into janicarts/vehicles, you 'climb onto them', so I don't see why you're so butthurt about one proc being overwritten for what is essentially an entirely different type of object

But yes, I did fix it. The bug where you can't buckle into janicarts is fixed in this PR, period. If you don't like the code, improve it, or do it yourself, but don't shill off your bugs onto other people and then complain and whine about the code that you had all the opportunity to not break.

RemieRichards commented:
>Uh you don't buckle into janicarts/vehicles

>I did fix it. The bug where you can't buckle into janicarts is fixed in this PR.

Contradictory statements.
and yes, you DO buckle into janicarts, they've always used copy pasted buckle code, you just brought that back, so do some fact checking.
Again I didn't break it, you just reverted it, if you'd have reported the bug to me, and said "X doesn't work, can you fix" instead of producing your retardo code then I'd have fixed it, no questions asked, and I'd have accepted it was my fault for not testing, but NO, you had to go be a stupid cunt.

paprka commented:
"Again I didn't break it"

Are you fucking stupid lol

>merge your update into PR
>buckling doesn't work

gee you sure told me

So you're basically saying you're denying the fact that you caused a bug because you're upset

10/10

RemieRichards commented:
If you'll kindly read, I said I'd accept the fact that I broke it, if you'd have brought it to my attention in a sane manner, you didn't so as far as I'm concerned Hitler himself could have caused that bug.

And are YOU fucking stupid, for yet again ignoring my points about the fact how this is buckling and how, your fix is not a fix but a band aid.

paprka commented:
ok please tell me how to fix it in a way that both retains the unique buckle message and the update_mob calling

I'm waiting!

RemieRichards commented:
override buckle_mob(), have it call ..(), add in the custom message, enjoy.
it may be a tad more specific than that, but that should work.

paprka commented:
if I call ..() won't it just display both buckle messages or are the messages at the buckle_mob level

RemieRichards commented:
buckle_mob() has no messages, user_buckle_mob() exists just for the (default) messages.
so calling ..() in buckle_mob() will handle most of the sanity checks and the actual buckling.

you'll also need
```
	if(!user.Adjacent(M) || user.restrained() || user.lying || user.stat)
		return
```
in the overriden buckle_mob(), above the ..() call

Yakk commented:
Did Remi just swear?

RemieRichards commented:
@Yakk while I may have once been unofficially called the nicest person in tgstation13 it's been atleast half a year since I started swearing openly around you guys. I have shit going on in my life and It stresses me out but I try not to swear.

Yakk commented:
I'm sorry. :(

RemieRichards commented:
@Yakk don't worry about it :)

paprka commented:
I'm not gonna add custom vars to your spaghetti procs remie

Commit:
remove copypaste

Commit:
Merge branch 'master' of https://github.com/tgstation/-tg-station into readdjanikart

Commit:
adds key spawning and stuff

paprka commented:
- Janitor spawns with janicart key to avoid grief
- Second janitor is shit out of luck and gets deluxe soap insted
- Added floor buffer upgrade to protolathe, replaced advanced mop
- Removed advanced mop
- Buffed normal mop slightly

stapler2025 commented:
Please make it locked to ID level access..getting griffed off your janicart back in the day got old, quick.

paprka commented:
It's a feature

Commit:
Merge branch 'master' of https://github.com/tgstation/-tg-station into readdjanikart

Conflicts:
	_maps/map_files/TgStation/tgstation.2.1.3.dmm

paprka commented:
conflicts fixd

Commit:
adds a check for the buffer upgrade

Miauw62 commented:
@RemieRichards is this good to merge?

RemieRichards commented:
@Miauw62 yeah it's fine.

Cheridan commented on code/game/jobs/job/civilian.dm:
```diff
@@ -209,6 +209,7 @@ Janitor
 	spawn_positions = 1
 	supervisors = "the head of personnel"
 	selection_color = "#dddddd"
+	var/global/lawyers = 1
```
EPIC COPYPASTA JOB, PAPRIKA :hurtrealbad:

paprka replied:
i'm'plyin'g i'm not procoder :laughing:

Miauw62 commented:
then merge it rr you're handling this pr :V

paprka commented:
i have to fix the copypaste anyway

paprka committed:
fixes copypaste in the job file
